docs/csharp/csharp-7.md

```diff
@@ -16,64 +16,539 @@ ms.assetid: fd41596d-d0c2-4816-b94d-c4d00a5d0243
+
+> [!NOTE]
+> The new features in Tuples require the `System.ValueTuple` type. For Visual Studio 15
+> Preview 5 and earlier preview releases, you must add the NuGet package "System.ValueTuple",
+> available in the pre-release stream.
+
+The `namedLetters` tuple contains fields referred to as `alpha` and
+`beta`. In a tuple assignment, you can also specify the names of the fields
+on the right hand side of the assignment:
+
+[!code-csharp[ImplicitNamedTuple](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#06_ImplicitNamedTuple "Implicitly named tuple")]
+
+The language allows you to specify names for the fields on both the
+left and right hand side of the assignment:
+
+[!code-csharp[NamedTupleConflict](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#07_NamedTupleConflict "Named tuple conflict")]
+
+The line above generates a warning, `CS8123`, telling you that the names on the right
+side of the assignment, `alpha` and `beta` are ignored because they conflict
+with the names on the left side, `first` and `second`.
+
+The examples above show the basic syntax to declare tuples. Tuples are
+most useful as return types for `private` and `internal` methods. Tuples
+provide a simple syntax for those methods to return multiple discrete values.
+You save the work of creating a `class` or a `struct` that
+defines the type returned. There is no need for creating a new symbol.
+
+Creating a tuple is more efficient and more productive.
+It is a simpler, lightweight syntax to define a data structure that carries
+more than one value. The example method below returns the minimimum and maximum
+values found in a sequence of integers:
+
+[!code-csharp[TupleReturningMethod](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#08_TupleReturningMethod "Tuple returning method")]
+
+The declaration for the method provides the names for the fields of the
+tuple that is returned. When you call the method, the return value is a 
+tuple whose fields are `max` and `min`:
+
+[!code-csharp[CallingTupleMethod](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#09_CallingTupleMethod "Calling a tuple returning method")]
+
+There may be times when you want to unpackage the members of a tuple that
+was returned from a method.  You can do that by declaring separate variables
+for each of the values in the tuple. This is called *deconstructing* the tuple:
+
+[!code-csharp[CallingWithDeconstructor](../../samples/snippets/csharp/new-in-7/new-in-7/program.cs#10_CallingWithDeconstructor "Deconstructing a tuple")]
+
+<!-- Add wildcards here, if they are in C# 7
+-->
 
-### Deconstruction
+You can also provide a simailar deconstruction for any type in .NET. This is
+done by writing a `Deconstruct` method as a member of the class. That
+`Deconstruct` method provides a set of `out` arguments for each of the
+properties you want to extract. Consider
+this `Point` class that provides a deconstructor method that extracts
```

Comment:
That's correct. The general term is 'deconstructor', but the method name is `Deconstruct`. 

Consistency FTW!